### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Replace Travis CI with GitHub Actions. Note: this temporarily disables e2e to get things working with GitHub actions and avoid slowing other PRs down for now (until e2e can be fully debugged).

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,162 @@
+name: Java CI
+
+on: [push, pull_request]
```
Comment (binh-dam-ibigroup) [inline .github/workflows/maven.yml:3]: We need to run this on the `push` event in order to generate builds for testing outside of the context of a pull request, right?

File: scripts/check-if-e2e-tests-should-run-on-ci.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Since the e2e tests take a while to run and it could present an inconvenience
+# to be making sure the e2e tests work on every single PR, only run the e2e
+# tests on CI for PRs to master or on commits directly to dev or master
+if [[ "$GITHUB_BASE_REF_SLUG" = "master" ]]; then
+  echo "SHOULD_RUN_E2E=true" >> $GITHUB_ENV && export SHOULD_RUN_E2E=true
+  echo 'Will run E2E tests because this is a PR to master'
+else
+  if [[ "$GITHUB_REPOSITORY" = "ibi-group/datatools-server" ]] && [[ "$GITHUB_REF_SLUG" = "master" || "$GITHUB_REF_SLUG" = "dev" || "$GITHUB_REF_SLUG" = "github-actions" ]]; then
```
Comment (binh-dam-ibigroup) [inline scripts/check-if-e2e-tests-should-run-on-ci.sh:8]: Can the two `if` blocks be combined? (The action is the same in both)

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, with one comment regarding checking whether a commit is to `master`.

**Review by br648 — APPROVED** [review]
LGTM!

File: scripts/check-if-e2e-tests-should-run-on-ci.sh (hunk 2 above)
Comment (landonreed) [reply scripts/check-if-e2e-tests-should-run-on-ci.sh:8]: The echoed message isn't the same though, so I'd prefer to keep them separate.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (landonreed) [reply .github/workflows/maven.yml:3]: Yes, I believe that's the case.

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+          cp "$FIRST_JAR" "deploy/dt-latest-$BRANCH_CLEAN.jar"
+      - name: Deploy to S3
+        uses: jakejarvis/s3-sync-action@master
+        with:
+          args: --acl public-read
+        env:
+          AWS_S3_BUCKET: datatools-builds
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          SOURCE_DIR: 'deploy'
```
Comment (evansiroky) [inline .github/workflows/maven.yml:162]: This is a security risk to give AWS credentials to a third party. I would have requested that this be changed if I had given a review.

**Pull request merged**